scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,260 @@ output:
 
-# integrated file containing samples to for DE 
+# integrated file containing all samples to use for DE analysis
 integrated_sce_file <- file.path(data_dir, "integrated", "rms_all_sce.rds")
 
-# sample metadata with information needed to setup DE
+# sample metadata with information needed to setup DE analysis
 sample_metadata_file <- file.path(data_dir, "annotations", "rms_sample_metadata.tsv")
 
-# output files 
+# directory to store output
 deseq_dir <- file.path("analysis", "rms", "deseq")
 if(!dir.exists(deseq_dir)){
   dir.create(deseq_dir, recursive = TRUE)
 }
 
-# results from deseq
+# results file to output from DE analysis
 deseq_output_file <- file.path(deseq_dir, "rms_myoblast_deseq_results.tsv")
 ```
 
+We can go ahead and read in the SCE object and the metadata file.
 
-```{r}
-# read in integrated SCE object 
+```{r read files}
+# read in the SCE object that has already been integrated
 integrated_sce <- readr::read_rds(integrated_sce_file)
 
 # read in sample metadata file 
 sample_metadata <- readr::read_tsv(sample_metadata_file)
 ```
 
-Before we set up our experiment, let's explore our dataset a little.
+## Dataset Exploration
 
-```{r}
-# First just plot by sample
-scater::plotReducedDim(integrated_sce,
-                       dimred = "fastmnn_UMAP",
-                       colour_by = "sample",
-                       point_size= 0.5,
-                       point_alpha = 0.2)
+Before we dive into differential expression, let's explore our integrated SCE object and the dataset a little more.
+
+We'll start by looking at what's inside the object.
+Here we should have both the original (uncorrected) data and the integrated (corrected) data for both the gene expression and the reduced dimensionality results.
+How are those stored in our object?
+
+```{r print sce}
+# print out entire object
+integrated_sce
+
+# look at the assay names in our object
+assayNames(integrated_sce)
+
+# look at the names of the dimensionality reduction present in the SCE object
+reducedDimNames(integrated_sce)
 ```
 
-We can also look at the sample metadata:
+When we look at the assay names we should see that there are 3 matrices, `counts`, `logcounts`, and `fastmnn_corrected`. 
+The `counts` and `logcounts` assays correspond to the uncorrected gene expression data that has been merged but NOT integrated.
+The `fastmnn_corrected` data contains the corrected gene expression data obtained from integration. 
+For this exercise we will not be using the `fastmnn_corrected` data (more on why not once we get to setting up the differential expression), but we need to be aware that it is present and be able to distinguish it from our uncorrected data. 
 
-```{r}
-sample_metadata
+In the `reducedDim` slots you should see `PCA` and `UMAP` which both correspond to the pre-integrated dimensionality reduction results.
+You should also see `fastmnn_PCA` and `fastmnn_UMAP` reduced dimensions, which correspond to the integrated results.
```

Comment:
Noting that my integration notebook is using (caps) `fastMNN_PCA` and `_UMAP`. 
Should I sync up w/ here?

Reply:
I think up to you on which one of us should switch? For some reason I really thought you were using all lower case but there is so much to keep track of.

Reply:
Weghing in in case this is not resolved, I would vote for `fastmnn_PCA`. I don't know why, except that I generally dislike mixed case for some reason.